src/Webserver.cc

```diff
@@ -13,95 +11,136 @@
-    std::string name = parent_name;
-    int count = inside_block;
+bool Webserver::load_configs(NginxConfig config) {
+    if (config.statements_.size() == 0) {
+        std::cerr << "Error: Empty config.\n";
+        return false;
+    }
 
     for (size_t i = 0; i < config.statements_.size(); i++) {
         std::shared_ptr<NginxConfigStatement> parent_statement = config.statements_[i];
+        NginxConfig child_config;
+        bool has_child = false;
 
         if (parent_statement->child_block_.get()) {
-            name = parent_statement->tokens_[0];
+            has_child = true;
+            child_config = *(parent_statement->child_block_.get());
         }
 
-        if (parent_statement->child_block_.get() && name != "" && count == 0) {
-            // Recurse on child block
-            bool success = load_configs(*(parent_statement->child_block_.get()), name, count + 1);
-            if (!success) {
-                return false;
+        size_t size = parent_statement->tokens_.size();
+        std::string first_token = "";
+        std::string second_token = "";
+        std::string third_token = "";
+
+        // Check statement size
+        if (size > 1 && size < 4) {
+            first_token = parent_statement->tokens_[0];
+            second_token = parent_statement->tokens_[1];
+            if (size == 3) {
+                third_token = parent_statement->tokens_[2];
             }
-            name = "";
-        }
-        // Parse statements
-        else if (parent_statement->tokens_.size() != 2) {
-            std::cerr << "Error: Invalid config syntax.\n";
-            std::cerr << parent_statement->ToString(1);
-            return false;
         }
         else {
-            if (name == "server") {
-                server_attributes[parent_statement->tokens_[0]] = parent_statement->tokens_[1];
+            return syntax_error(parent_statement);
+        }
+
+        // Parse statements
+        if (first_token == "port" && third_token == "") {
+            if (has_child) {
+                return syntax_error(parent_statement);
             }
-            else if (name == "directories") {
-                dir_attributes[parent_statement->tokens_[0]] = parent_statement->tokens_[1];
+
+            bool is_number = (second_token.find_first_not_of("1234567890") == std::string::npos);
+            if (is_number) {
+                port = std::stoi(second_token);
             }
             else {
-                //ignore statements not in a child block
-                ;
+                return syntax_error(parent_statement);
             }
         }
+        else if (first_token == "path" && third_token != "") {
+                if (!add_handler(second_token, child_config, third_token)) {
+                    return false;
+                }
+        }
+        else if (first_token == "default" && second_token == "NotFoundHandler" && third_token == "") {
```

Comment:
Does the default have to be NotFoundHandler? Maybe we should just leave default to be whatever the config specifies, and if there is no specification, then we add NotFoundHandler as default?